### PR TITLE
Widens TotalRideValue to fix overflow reported in #19292

### PR DESCRIPTION
**Pull request by eaeiv**
Widens all instances of TRV from money16 to money64. 
Fixes OpenRCT2/OpenRCT2/#19292

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
You'll need to increment the park file minimum compatible to widen the type. `PARK_FILE_MIN_VERSION` and `PARK_FILE_CURRENT_VERSION` should be set to 17.

**Comment by duncanspumpkin** [comment]
You'll also need to add logic within parkfile.cpp to convert the old size to the new size on versions < 17.

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
See comments. After making the parkfile changes it will pass the replays.

**Review by duncanspumpkin — APPROVED** [review]

**Review by Broxzier — CHANGES_REQUESTED** [review]
Please add a changelog entry for this fix. The changelog is located in the distrubution folder. It's ordered by type (fix in this case) and issue number.

File: src/openrct2/world/Park.cpp [hunk 1]
```diff
@@ -532,7 +532,7 @@ money16 Park::CalculateTotalRideValueForMoney() const
         // Add ride value
         if (ride.value != RIDE_VALUE_UNDEFINED)
         {
-            money16 rideValue = static_cast<money16>(ride.value);
+            money64 rideValue = static_cast<money64>(ride.value);
```
Comment (CookiePLMonster) [inline src/openrct2/world/Park.cpp:535]: Isn't this case redundant now?

Comment (duncanspumpkin) [reply src/openrct2/world/Park.cpp:535]: It's still going from a uint16 to an int64

**Pull request merged**